### PR TITLE
Calculate and set leverage for derivatives orders.

### PR DESCRIPTION
**Pull request by tekwani**
Calculate and set leverage for derivatives orders.
Submit stop loss order if enabled.
Cancel stop loss order on position close.
Close position on stop loss order filled.

### CODE REVIEW

File: lib/constants.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (tarcisiozf) [inline lib/constants.js:1]: add 'use strict'

File: lib/updates/on_order.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Called for a closed order
+ *
+ * @param {Object} state
+ * @param {Object} order
+ * @return {Promise} p - resolves to nextState
+ */
+const onOrder = async (state = {}, order = {}) => {
+  // check if stop order linked to the position is filled
+  if (order.type === STOP_ORDER_TYPE && order.amount === 0) {
```
Comment (tarcisiozf) [inline lib/updates/on_order.js:20]: I suggest using "early return" to those if statements

Comment (tekwani) [reply lib/updates/on_order.js]: Fixed.

File: lib/constants.js (hunk 1 above)
Comment (tekwani) [reply lib/constants.js:3]: Fixed.

**Review by tarcisiozf — APPROVED** [review]

**Pull request merged**